src/lib/utils/string_utils.cpp

```diff
@@ -52,4 +55,25 @@ std::string replace_addresses(const std::string& input) {
   return std::regex_replace(input, std::regex{"0x[0-9A-Fa-f]{4,}"}, "0x00000000");
 }
 
+void table_key_constraints_to_stream(std::ostream& stream, const std::shared_ptr<const Table>& table,
+                                     const std::string& separator) {
+  const auto& table_key_constraints = table->soft_key_constraints();
+  for (auto constraint_it = table_key_constraints.cbegin(); constraint_it != table_key_constraints.cend();
```

Comment:
I guess your version is fine, too. I'd just add `Assert`s that neither the constraints nor each constraint's columns are empty to prevent segfaults when dereferencing the decreased iterators.

Reply:
Anyways, is this method well-placed in `string_utils` or do yo prefer it to be a member of `Table`?

Reply:
As discussed in person: creating `print_utils.hpp/cpp` files for this function might be an option. Then, we might also move other print-supportive code to the print_utils files (not as part of this PR).